cli/assets/templates/zig/build.zig

```diff
@@ -1,18 +1,22 @@
 const std = @import("std");
 
-pub fn build(b: *std.build.Builder) !void {
-    const mode = b.standardReleaseOptions();
-    const lib = b.addSharedLibrary("cart", "src/main.zig", .unversioned);
+pub fn build(b: *std.Build) !void {
+    const optimize = b.standardOptimizeOption(.{});
+
+    const lib = b.addSharedLibrary(.{
+        .name = "cart",
+        .root_source_file = .{ .path = "src/main.zig" },
+        .target = .{ .cpu_arch = .wasm32, .os_tag = .freestanding },
+        .optimize = optimize,
+    });
 
-    lib.setBuildMode(mode);
-    lib.setTarget(.{ .cpu_arch = .wasm32, .os_tag = .freestanding });
     lib.import_memory = true;
     lib.initial_memory = 65536;
     lib.max_memory = 65536;
     lib.stack_size = 14752;
 
     // Export WASM-4 symbols
-    lib.export_symbol_names = &[_][]const u8{ "start", "update" };
+    lib.export_symbol_names = &[_][]const u8{ "update" };
```

Comment:
Is removing "start" here intentional?

Reply:
Yes, I got a compile error with it included. `start` is already exported by default.

Reply:
Ok I've done some research and here's what I've found:
1. We could either do it the current way `lib.export_symbol_names = &[_][]const u8{ "start", "update" };` and require the user to have `export fn start() void {}` in their  code if they have no `start` code.
2. We could do `lib.rdynamic = true;` which will actually include all exported functions in the Wasm binary. I think this is the best approach because it doesn't require the user to have `export fn start() void {}` in their code or even know about the existence of `start` to make their game run. I don't think it'll be a problem to always include all functions that are explicitly marked as `export` in the Zig code.

Reply:
Requiring the `export` keyword on the `start` and `update` functions sounds fine to me.

Reply:
Done